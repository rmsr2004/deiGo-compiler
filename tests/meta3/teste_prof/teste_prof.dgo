package main;
func g() int {}
func j() bool {}
func f() {
    return -j();
    if (g()) {}
    for (g()) {}
    return g();
}