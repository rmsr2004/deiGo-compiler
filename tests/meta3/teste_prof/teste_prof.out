Line 5, column 9: Operator - cannot be applied to type bool
Line 5, column 9: Incompatible type undef in return statement
Line 6, column 6: Incompatible type int in if statement
Line 7, column 7: Incompatible type int in for statement
Line 8, column 9: Incompatible type int in return statement
===== Global Symbol Table =====
g    ()    int
j    ()    bool
f    ()    none

===== Function g() Symbol Table =====
return        int

===== Function j() Symbol Table =====
return        bool

===== Function f() Symbol Table =====
return        none

Program
..FuncDecl
....FuncHeader
......Identifier(g)
......Int
......FuncParams
....FuncBody
..FuncDecl
....FuncHeader
......Identifier(j)
......Bool
......FuncParams
....FuncBody
..FuncDecl
....FuncHeader
......Identifier(f)
......FuncParams
....FuncBody
......Return
........Minus - undef
..........Call - bool
............Identifier(j) - ()
......If
........Call - int
..........Identifier(g) - ()
........Block
........Block
......For
........Call - int
..........Identifier(g) - ()
........Block
......Return
........Call - int
..........Identifier(g) - ()